src/components/ugas/GasStats.vue

```diff
@@ -0,0 +1,518 @@
+import { OutsideClick } from "../../utils/directives/outside-click";
+
+export default {
+  directives: {
+    OutsideClick,
+  },
+  data() {
+    return {
+      showDateInput: false,
+      showPopup: false,
+      currency: true,
+      interval: "",
+      badgeState: "All Time",
+      txGasCostETH: "",
+      txGasCostUSD: "",
+      averageTxPrice: "",
+      txCount: "",
+      failedTxCount: "",
+      failedTxGasCostETH: "",
+      failedTxGasCostUSD: "",
+      inputStartDate: "",
+      inputEndDate: "",
+    };
+  },
+  props: {
+    size: Number,
+  },
+  computed: {
+    style() {
+      return null;
+    },
+  },
+  methods: {
+    ...mapActions(["getUserTxStats"]),
+    async onClose() {
+      this.showPopup = false;
+    },
+    async updateBadge(value) {
+      this.badgeState = value;
+    },
+    async setLoadingState() {
+      this.txGasCostETH = "...";
+      this.txGasCostUSD = "...";
+      this.averageTxPrice = "...";
+      this.txCount = "...";
+      this.failedTxCount = "...";
+      this.failedTxGasCostETH = "...";
+      this.failedTxGasCostUSD = "...";
+    },
+    async updateInterval(value) {
+      this.badgeState = value;
+      this.interval = value;
+      this.inputStartDate = null;
+      this.inputEndDate = null;
+      this.showDateInput = false;
+      await this.getAccountStats();
+    },
+    async updateCustomDate(value) {
+      this.badgeState = value;
+      this.interval = "";
+      await this.getAccountStats();
+    },
+    async getAccountStats() {
+      await this.setLoadingState();
+      const price = await getOffchainPriceFromTokenSymbol("uUSDrETH");
+      const [txGasCostETH, averageTxPrice, txCount, failedTxCount, failedTxGasCostETH] = await this.getUserTxStats({
+        interval: this.interval,
+        startDate: this.inputStartDate,
+        endDate: this.inputEndDate,
+      });
+
+      this.txGasCostETH = txGasCostETH;
+      this.txGasCostUSD = new BigNumber(txGasCostETH / price).decimalPlaces(0);
+      this.averageTxPrice = averageTxPrice;
+      this.txCount = txCount;
+      this.failedTxCount = failedTxCount;
+      this.failedTxGasCostETH = failedTxGasCostETH;
+      this.failedTxGasCostUSD = new BigNumber(failedTxGasCostETH / price).decimalPlaces(0);
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+.hidden {
+  display: none;
+}
+ul {
+  list-style: none;
+  margin: 0;
+  padding: 0;
+}
+.toggle-grey {
+  background-color: rgba(229, 231, 235, 1);
+}
+.toggle-blue {
+  background: rgba(199, 167, 249, 1);
+}
+.grey {
+  color: rgba(75, 85, 99, 1);
+}
+.green {
+  color: rgba(52, 211, 153, 1);
+}
+.primary {
+  color: var(--primary);
+}
+.dropdown-list {
+  max-height: 25rem;
+  border-radius: 0.375rem;
+  padding-top: 0.25rem;
+  padding-bottom: 0.25rem;
+  font-size: 1rem;
+  line-height: 1.5rem;
+  overflow: auto;
+}
+.dropdown-list li:hover {
+  background-color: rgba(229, 231, 235, 1);
```

Comment:
thats for the moment, later these must be globally defined
```suggestion
  background-color: #f2eef0;
```